contracts/core/EntryPoint.sol

```diff
@@ -521,6 +512,33 @@ contract EntryPoint is IEntryPoint, StakeManager, NonceManager, ReentrancyGuardT
         }
     }
 
+    // call sender.validateUserOp()
+    // handle wrong output size with FailedOp
+    function _callValidateUserOp(PackedUserOperation calldata op, UserOpInfo memory opInfo, uint256 missingAccountFunds, uint256 opIndex)
+    internal returns (uint256 validationData) {
+        uint256 saveFreePtr;
+        assembly ("memory-safe") {
+            saveFreePtr := mload(0x40)
+        }
```

Comment:
I looked into it, and there is probably no way to have a clean generic function that wraps this 'save-restore free memory pointer' trick.
Maybe for simplicity we could create inline `save()` and `restore()` functions and reuse those instead?
These assembly blocks kind of pollute the code.